### ISSUE TITLE
Tensor.eq with nan does not match PyTorch on LLVM backend

### ISSUE THREAD

**Issue by jla524** [tinygrad/tinygrad#1069]
Steps to reproduce:
```
jacky@ephphatha tinygrad % METAL=1 python3
Python 3.11.3 (main, Apr  7 2023, 20:13:31) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from tinygrad.tensor import Tensor
>>> x = Tensor.full((3, 3), float("nan"))
>>> (x != x).numpy()
array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]], dtype=float32)
>>> import torch
>>> x = torch.full((3, 3), float("nan"), device="mps")
>>> x != x
tensor([[True, True, True],
        [True, True, True],
        [True, True, True]], device='mps:0')
```

Edit: LLVM also returns all zeros

**Comment by dc-dc-dc** [comment]
No longer an issue on Metal but still getting 0 on LLVM backend  

Metal Output: 
```python
>>> from tinygrad.tensor import Tensor
>>> x = Tensor.full((3, 3), float("nan"))
>>> (x != x).numpy()
array([[1., 1., 1.],
       [1., 1., 1.],
       [1., 1., 1.]], dtype=float32)
>>> x
<Tensor <LB (3, 3) dtypes.float op=const<nan, dtypes.float> st=ShapeTracker(shape=(3, 3), views=[View(shape=(3, 3), strides=(0, 0), offset=0, mask=None, contiguous=False, shape_strides=((9, 0),))])> on METAL with grad None>
```

LLVM Output:
```python
>>> from tinygrad.tensor import Tensor
>>> x = Tensor.full((3, 3), float("nan"))
>>> x
<Tensor <LB (3, 3) dtypes.float op=MovementOps.EXPAND st=ShapeTracker(shape=(3, 3), views=[View(shape=(3, 3), strides=(0, 0), offset=0, mask=None, contiguous=False, shape_strides=((9, 0),))])> on LLVM with grad None>
>>> (x != x).numpy()
array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]], dtype=float32)
```

**Comment by chenyuxyz** [comment]
fixed on master. LLVM fixed in #5282
